instrumentation/elasticsearch/lib/opentelemetry/instrumentation/elasticsearch/patches/client.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'uri'
+
+module OpenTelemetry
+  module Instrumentation
+    module Elasticsearch
+      module Patches
+        # Module to prepend to Elasticsearch::Client for instrumentation
+        module Client
+          def perform_request(*args)
+            response = nil
+
+            tracer.in_span(
+              'elasticsearch.query',
+              attributes: tracing_attributes(*args),
+              kind: :client
+            ) do |span|
+              response = super(*args)
+              span.set_attribute('http.status_code', response.status)
+            end
+            response
+          end
+
+          private
+
+          def tracing_attributes(*args)
+            method = args[0]
+            path = args[1]
+            params = args[2]
+            body = args[3]
+            url = URI.parse(path).path
+
+            params = JSON.generate(params) if params && !params.is_a?(String)
+            body = JSON.generate(body) if body && !body.is_a?(String)
+
+            {
+              'elasticsearch.url' => url,
+              'elasticsearch.method' => method,
+              'elasticsearch.params' => params,
+              'elasticsearch.body' => body || ''
+            }.merge(connection_attributes)
```

Comment:
You might consider using the [http](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md) or [rpc](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md) conventions here, but I'm not sure that either one really applies (what do you think @mmontalvo ?)

Reply:
I think we need to consider the http one 👍